src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -219,6 +217,47 @@ func (client *Client) proxiedDialer(orig func(network, addr string) (net.Conn, e
 	}
 }
 
+func (client *Client) dialCONNECT(addr string, port int) (net.Conn, error) {
```

Comment:
Centralized this logic in one function for use both in HTTP proxy and SOCKS proxy.